### PR TITLE
Refactor occurences of `f = open(...)` using `with open(...) as f` instead

### PR DESCRIPTION
**Pull request by maldil**
## Description
Utilizing Python `with` statement to make the code more Pythonic

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: skimage/io/tests/test_sift.py [hunk 1]
```diff
@@ -47,12 +46,12 @@ def test_load_surf():
...
-68.5773 61.474 0.0313267 0 0.0313267 1 -0.10198 0.130987 -0.0321845 0.0487543 -0.0900435 0.121113 -0.100917 0.0444702 -0.0151742 0.107604 -0.0542035 0.014069 -0.00594097 0.0339933 -0.00994295 0.0127262 -0.125613 0.192551 -0.0174399 0.0433488 -0.272698 0.164641 -0.0676735 0.0467444 -0.0527907 0.258005 -0.0818114 0.0440569 -0.0104433 0.0548934 -0.0323454 0.0145296 -0.112357 0.199223 -0.0532903 0.0332622 -0.342481 0.207469 -0.0526129 0.0741355 -0.256234 0.402708 -0.108296 0.117362 -0.0560274 0.128856 -0.123509 0.0510046 -0.0198793 0.0775934 -0.103863 0.00406679 -0.10264 0.1312 -0.108244 0.0812913 -0.127868 0.182924 -0.0680942 0.071913 -0.0858004 0.144806 -0.0176522 0.0686146''')
-    f.close()
+    with open(fname, 'wb') as f:
```
Comment (grlee77) [inline skimage/io/tests/test_sift.py:49]: There is one more open on line 59 that could also be converted to a context manager

I see there was some whitespace change to what is written, but if the tests are passing then apparently load_surf

File: skimage/io/tests/test_sift.py [hunk 2]
```diff
@@ -47,12 +46,12 @@ def test_load_surf():
...
+        68.5773 61.474 0.0313267 0 0.0313267 1 -0.10198 0.130987 -0.0321845 0.0487543 -0.0900435 0.121113 -0.100917 0.0444702 -0.0151742 0.107604 -0.0542035 0.014069 -0.00594097 0.0339933 -0.00994295 0.0127262 -0.125613 0.192551 -0.0174399 0.0433488 -0.272698 0.164641 -0.0676735 0.0467444 -0.0527907 0.258005 -0.0818114 0.0440569 -0.0104433 0.0548934 -0.0323454 0.0145296 -0.112357 0.199223 -0.0532903 0.0332622 -0.342481 0.207469 -0.0526129 0.0741355 -0.256234 0.402708 -0.108296 0.117362 -0.0560274 0.128856 -0.123509 0.0510046 -0.0198793 0.0775934 -0.103863 0.00406679 -0.10264 0.1312 -0.108244 0.0812913 -0.127868 0.182924 -0.0680942 0.071913 -0.0858004 0.144806 -0.0176522 0.0686146''')
```
Comment (grlee77) [inline skimage/io/tests/test_sift.py:53]: avoid introducing extra whitespace to the descriptor file

```suggestion
2
38.3727 62.0491 0.0371343 0 0.0371343 -1 -0.0705589 0.0130983 -0.00460534 0.132168 -0.0718833 0.0320583 -0.0134032 0.0988654 -0.0542241 0.0171002 -0.00135754 0.105755 -0.0362088 0.0151748 -0.00694272 0.0610017 -0.247091 0.109605 -0.0337623 0.0813307 -0.24185 0.278548 -0.0494523 0.107804 -0.166312 0.0691584 -0.0288199 0.138476 -0.110956 0.0280772 -0.0752509 0.0736344 -0.22667 0.164226 -0.0544717 0.0388139 -0.30194 0.33065 -0.0537507 0.0596398 -0.245395 0.110925 -0.0603322 0.0239389 -0.18726 0.0374145 -0.0355872 0.0140762 -0.129022 0.135104 -0.0703396 0.0374049 -0.24256 0.222544 -0.0536354 0.0501252 -0.209004 0.0971316 -0.0550094 0.0229767 -0.125547 0.0317879 -0.0291574 0.0124569
68.5773 61.474 0.0313267 0 0.0313267 1 -0.10198 0.130987 -0.0321845 0.0487543 -0.0900435 0.121113 -0.100917 0.0444702 -0.0151742 0.107604 -0.0542035 0.014069 -0.00594097 0.0339933 -0.00994295 0.0127262 -0.125613 0.192551 -0.0174399 0.0433488 -0.272698 0.164641 -0.0676735 0.0467444 -0.0527907 0.258005 -0.0818114 0.0440569 -0.0104433 0.0548934 -0.0323454 0.0145296 -0.112357 0.199223 -0.0532903 0.0332622 -0.342481 0.207469 -0.0526129 0.0741355 -0.256234 0.402708 -0.108296 0.117362 -0.0560274 0.128856 -0.123509 0.0510046 -0.0198793 0.0775934 -0.103863 0.00406679 -0.10264 0.1312 -0.108244 0.0812913 -0.127868 0.182924 -0.0680942 0.071913 -0.0858004 0.144806 -0.0176522 0.0686146''')
```

**Review by grlee77 — CHANGES_REQUESTED** [review]
Hi @maldil, this looks like a good change. Please see the minor comments.

**Comment by alexdesiqueira** [comment]
Thank you @maldil! Looks good to me, but I'll wait for the last word of @grlee77.
Thanks again!

**Comment by rfezzani** [comment]
Thank you @maldil for this PR!
Usinging `grep -r "\.close(" skimage/`, I found

```
skimage/io/tests/test_sift.py:    f.close()
skimage/io/tests/test_sift.py:    f.close()
skimage/io/tests/test_sift.py:    f.close()
skimage/io/tests/test_sift.py:    f.close()
skimage/io/tests/test_sift.py:    f.close()
skimage/io/tests/test_simpleitk.py:        f.close()
skimage/io/tests/test_tifffile.py:        f.close()
skimage/io/tests/test_pil.py:    f.close()
skimage/io/tests/test_pil.py:    f.close()
skimage/io/tests/test_pil.py:    f.close()
skimage/io/tests/test_pil.py:        img.close()
skimage/io/tests/test_imageio.py:        f.close()
skimage/io/tests/test_imageio.py:        f.close()
skimage/io/tests/test_imread.py:        f.close()
skimage/io/collection.py:                    im.fp.close()
skimage/io/_plugins/fits_plugin.py:    hdulist.close()
skimage/io/_plugins/fits_plugin.py:        hdulist.close()
skimage/io/_plugins/fits_plugin.py:    hdulist.close()
skimage/io/_plugins/pil_plugin.py:        image.fp.close()
skimage/io/sift.py:        f.close()
skimage/_shared/_tempfile.py:    tempfile_stream.close()
skimage/_shared/testing.py:    temp_file.close()
```

There is potentially some more places in the package where this refactoring can be applied :wink:

**Comment by rfezzani** [comment]
Hi @maldil, thank you again for your time. The general scope of your PR is about using the context manager. The trickiest refactoring is the `io/sift.py` one, but all the others are "straightforward"... Let's wait for @grlee77 point of view :wink:

**Comment by alexdesiqueira** [comment]
@rfezzani I think this one is good to go, and the others can be dealt with later.
Thank you for your work so far, @maldil!

**Comment by rfezzani** [comment]
Well, as you wish. No problem, I opened a PR to complete this. Thank you again @maldil

**Review by alexdesiqueira — APPROVED** [review]

**Review by mkcor — APPROVED** [review]

**Review by grlee77 — APPROVED** [review]
Thanks, @. This looks good now. Sorry for taking so long to get back to it!

**Pull request merged**